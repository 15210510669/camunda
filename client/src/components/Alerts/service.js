import {get, post, del, put} from 'request';

export async function loadAlerts() {
  const response = await get('/api/alert');

  return await response.json();
}

export async function loadReports() {
  const response = await get('/api/report');

  return await response.json();
}

export async function saveNewAlert(alert) {
  const response = await post('/api/alert', alert);

  const json = await response.json();

  return json.id;
}

export async function updateAlert(id, alert) {
  return await put('/api/alert/' + id, alert);
}

export async function deleteAlert(id) {
  return await del('api/alert/' + id);
}

