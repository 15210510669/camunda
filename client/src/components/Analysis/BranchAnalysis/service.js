/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {post} from 'request';

export async function loadFrequencyData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  const response = await post(
    'api/report/evaluate',
    createFlowNodeFrequencyReport(
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter
    )
  );

  return await response.json();
}

function createFlowNodeFrequencyReport(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  return {
    combined: false,
    reportType: 'process',
    data: {
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter,
      view: {
        entity: 'flowNode',
        property: 'frequency',
      },
      groupBy: {
        type: 'flowNodes',
        unit: null,
      },
      visualization: 'heat',
    },
  };
}

export async function loadCorrelationData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter,
  gateway,
  end
) {
  const response = await post('api/analysis/correlation', {
    processDefinitionKey,
    processDefinitionVersions,
    tenantIds,
    filter,
    gateway,
    end,
  });

  return await response.json();
}
