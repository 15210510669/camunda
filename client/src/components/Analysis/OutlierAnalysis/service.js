/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';

export async function loadNodesOutliers(config) {
  const response = await get('api/analysis/flowNodeOutliers', config);
  return await response.json();
}

export {loadDurationData, loadCommonOutliersVariables, getInstancesDownloadUrl} from './service.ts';
