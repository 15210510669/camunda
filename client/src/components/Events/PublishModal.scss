.PublishModal {
  p {
    color: #626262;
  }

  .permission {
    h4 {
      font-weight: bold;
      margin-bottom: 10px;
    }

    .Button {
      margin-left: 10px;
    }
  }
}
