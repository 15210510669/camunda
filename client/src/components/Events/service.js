/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get, post, put, del} from 'request';

export {getUsers, updateUsers, publish} from './service.ts';

export async function loadProcesses() {
  const response = await get('api/eventBasedProcess');
  return await response.json();
}

export async function loadProcess(id) {
  const response = await get('api/eventBasedProcess/' + id);
  return await response.json();
}

export async function removeProcess(id) {
  return await del(`api/eventBasedProcess/${id}`);
}

export async function cancelPublish(id) {
  return await post(`api/eventBasedProcess/${id}/_cancelPublish`);
}

export async function createProcess(payload) {
  const response = await post('api/eventBasedProcess', payload);
  const json = await response.json();

  return json.id;
}

export async function updateProcess(id, name, xml, mappings, eventSources) {
  return await put('api/eventBasedProcess/' + id, {name, xml, mappings, eventSources});
}

export async function loadEvents(body, searchTerm, sorting) {
  const query = {};
  if (searchTerm) {
    query.searchTerm = searchTerm;
  }

  if (sorting) {
    query.sortBy = sorting.by;
    query.sortOrder = sorting.order;
  }

  const response = await post('api/event/count', body, {query});
  return await response.json();
}

export async function getCleanedMappings(body) {
  const response = await post('api/eventBasedProcess/_mappingCleanup', body);
  return await response.json();
}

export function isNonTimerEvent(node) {
  if (!node?.$instanceOf('bpmn:Event')) {
    return false;
  }

  if (node.eventDefinitions) {
    return !node.eventDefinitions[0].$instanceOf('bpmn:TimerEventDefinition');
  }

  return true;
}

export async function loadIngestedEvents(params) {
  const response = await get('api/event', params);

  return await response.json();
}

export async function deleteEvents(eventsIds) {
  return await post('api/event/delete', eventsIds);
}

export async function loadExternalGroups(query) {
  const response = await get('api/event/groups', query);

  return await response.json();
}

export async function deleteProcesses(processes) {
  return await post(
    'api/eventBasedProcess/delete',
    processes.map(({id}) => id)
  );
}
