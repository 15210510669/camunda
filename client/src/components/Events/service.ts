/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get, post, put} from 'request';

import {User} from 'components';

export async function getUsers(id: string): Promise<User[]> {
  const response = await get(`api/eventBasedProcess/${id}/role`);
  return await response.json();
}

export async function updateUsers(id: string, newUsers: User[]) {
  return await put(`api/eventBasedProcess/${id}/role`, newUsers);
}

export async function publish(id: string) {
  return await post(`api/eventBasedProcess/${id}/_publish`);
}

export async function loadExternalGroups(query?: Record<string, unknown>) {
  const response = await get('api/event/groups', query);

  return await response.json();
}
