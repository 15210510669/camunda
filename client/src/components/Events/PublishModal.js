/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';

import {Modal, Button} from 'components';
import {t} from 'translation';
import {withErrorHandling, withUser} from 'HOC';
import {showError, addNotification} from 'notifications';

import {publish, getUsers} from './service';
import UsersModal from './UsersModal';
import './PublishModal.scss';

export class PublishModal extends React.Component {
  state = {
    loading: false,
    editingAccess: null,
    isPrivate: false,
  };

  componentDidMount() {
    this.props.mightFail(getUsers(this.props.id), (users) => this.updatePrivate(users), showError);
  }

  updatePrivate = (users) => {
    this.setState({isPrivate: users.length === 1 && users[0].identity.id === this.props.user?.id});
  };

  publish = () => {
    const {mightFail, id, onPublish, onClose} = this.props;
    this.setState({loading: true});
    mightFail(
      publish(id),
      () => {
        this.setState({loading: false});
        addNotification({type: 'hint', text: t('events.publishStart')});
        onPublish();
        onClose();
      },
      (error) => {
        this.setState({loading: false});
        showError(error);
      }
    );
  };

  closeUsersModal = (users) => {
    this.setState({editingAccess: null});
    if (users) {
      this.updatePrivate(users);
    }
  };

  render() {
    const {id, onClose, republish} = this.props;
    const {loading, editingAccess, isPrivate} = this.state;

    return (
      <Modal open={id} onClose={onClose} onConfirm={this.publish} className="PublishModal">
        <Modal.Header>
          {republish ? t('events.publishModal.republishHead') : t('events.publishModal.head')}
        </Modal.Header>
        <Modal.Content>
          {republish ? (
            <p>{t('events.publishModal.republishText')}</p>
          ) : (
            <p>{t('events.publishModal.text')}</p>
          )}
          <div className="permission">
            <h4>{t('events.permissions.whoHasAccess')}</h4>
            {isPrivate ? t('events.permissions.private') : t('events.permissions.userGranted')}
            <Button onClick={() => this.setState({editingAccess: id})} link>
              {t('common.change')}...
            </Button>
          </div>
          {editingAccess && <UsersModal id={editingAccess} onClose={this.closeUsersModal} />}
        </Modal.Content>
        <Modal.Actions>
          <Button main disabled={loading} className="close" onClick={onClose}>
            {t('common.cancel')}
          </Button>
          <Button main disabled={loading} primary className="confirm" onClick={this.publish}>
            {t(`events.publish`)}
          </Button>
        </Modal.Actions>
      </Modal>
    );
  }
}

export default withUser(withErrorHandling(PublishModal));
