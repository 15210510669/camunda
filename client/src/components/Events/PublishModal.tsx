/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useState, useEffect} from 'react';
import {Button} from '@carbon/react';

import {Modal, Button as LegacyButton, User} from 'components';
import {t} from 'translation';
import {withErrorHandling, WithErrorHandlingProps, withUser, WithUserProps} from 'HOC';
import {showError, addNotification} from 'notifications';

import {publish, getUsers} from './service';
import UsersModal from './UsersModal';

import './PublishModal.scss';

interface PublishModalProps extends WithErrorHandlingProps, WithUserProps {
  id: string;
  republish: boolean;
  onClose: () => void;
  onPublish: (id?: string) => void;
}

export function PublishModal({
  onClose,
  onPublish,
  mightFail,
  id,
  user,
  republish,
}: PublishModalProps): JSX.Element {
  const [loading, setLoading] = useState<boolean>(false);
  const [editingAccess, setEditingAccess] = useState<string | null>(null);
  const [isPrivate, setIsPrivate] = useState<boolean>(false);

  useEffect(() => {
    mightFail(
      getUsers(id),
      (users: User[]) => {
        setIsPrivate(users.length === 1 && users[0]?.identity.id === user?.id);
      },
      showError
    );
  }, [id, mightFail, user?.id]);

  const publishProcess = () => {
    setLoading(true);
    mightFail(
      publish(id),
      () => {
        addNotification({type: 'hint', text: t('events.publishStart')});
        onPublish();
        onClose();
      },
      (error) => {
        showError(error);
      },
      () => setLoading(false)
    );
  };

  const closeUsersModal = (users?: User[] | null) => {
    setEditingAccess(null);
    if (users) {
      setIsPrivate(users.length === 1 && users[0]?.identity.id === user?.id);
    }
  };

  return (
    <Modal open={!!id} onClose={onClose} className="PublishModal">
      <Modal.Header>
        {republish ? t('events.publishModal.republishHead') : t('events.publishModal.head')}
      </Modal.Header>
      <Modal.Content>
        {republish ? (
          <p>{t('events.publishModal.republishText')}</p>
        ) : (
          <p>{t('events.publishModal.text')}</p>
        )}
        <div className="permission">
          <h4>{t('events.permissions.whoHasAccess')}</h4>
          {isPrivate ? t('events.permissions.private') : t('events.permissions.userGranted')}
          <LegacyButton onClick={() => setEditingAccess(id)} link>
            {t('common.change')}...
          </LegacyButton>
        </div>
        {editingAccess && <UsersModal id={editingAccess} onClose={closeUsersModal} />}
      </Modal.Content>
      <Modal.Footer>
        <Button kind="secondary" disabled={loading} className="close" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button disabled={loading} className="confirm" onClick={publishProcess}>
          {t(`events.publish`)}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}

export default withUser(withErrorHandling(PublishModal));
