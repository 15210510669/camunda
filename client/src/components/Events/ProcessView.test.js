/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {Deleter} from 'components';

import {removeProcess, loadProcess, cancelPublish} from './service';
import PublishModal from './PublishModal';

import ProcessViewWithErrorHandling from './ProcessView';

const ProcessView = ProcessViewWithErrorHandling.WrappedComponent;

jest.mock('./service', () => ({
  removeProcess: jest.fn(),
  loadProcess: jest.fn().mockReturnValue({
    id: 'processId',
    name: 'Process Name',
    xml: 'Process XML',
    mappings: {},
    state: 'mapped'
  }),
  cancelPublish: jest.fn()
}));

const props = {
  id: 'processId',
  mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
  onDelete: jest.fn()
};

it('should match snapshot', () => {
  const node = shallow(<ProcessView {...props} />);

  expect(node).toMatchSnapshot();
});

it('should load process by id', () => {
  shallow(<ProcessView {...props} />);

  expect(loadProcess).toHaveBeenCalledWith('processId');
});

it('should pass entity to Deleter', () => {
  const node = shallow(<ProcessView {...props} />);

  node.find('.delete-button').simulate('click');

  expect(node.find(Deleter).prop('entity')).toEqual({id: 'processId', name: 'Process Name'});
});

it('should call the onDelete callback', () => {
  const node = shallow(<ProcessView {...props} />);

  node.find(Deleter).prop('onDelete')();

  expect(props.onDelete).toHaveBeenCalled();
});

it('should allow deletion of a process', () => {
  const node = shallow(<ProcessView {...props} />);

  node.find(Deleter).prop('deleteEntity')({id: '1'});

  expect(removeProcess).toHaveBeenCalledWith('1');
});

it('should pass a process id to the PublishModal', () => {
  const node = shallow(<ProcessView {...props} />);

  node.find('.publish-button').simulate('click');

  expect(node.find(PublishModal).prop('id')).toBe('processId');
  expect(node.find(PublishModal).prop('republish')).toBe(false);
});

it('should correctly set the republish prop on the PublishModal', () => {
  loadProcess.mockReturnValueOnce({
    id: 'processId',
    name: 'Process Name',
    xml: 'Process XML',
    mappings: {},
    state: 'unpublished_changes'
  });
  const node = shallow(<ProcessView {...props} />);

  node.find('.publish-button').simulate('click');

  expect(node.find(PublishModal).prop('republish')).toBe(true);
});

it('should allow cancel of an ongoing publish', () => {
  loadProcess.mockReturnValueOnce({
    id: 'processId',
    name: 'Process Name',
    xml: 'Process XML',
    mappings: {},
    state: 'publish_pending',
    publishingProgress: 14
  });

  const node = shallow(<ProcessView {...props} />);

  node.find('.cancel-button').simulate('click');

  expect(cancelPublish).toHaveBeenCalledWith('processId');
});
