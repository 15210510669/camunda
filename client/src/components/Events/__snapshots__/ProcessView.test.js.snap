// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="ProcessView"
>
  <div
    className="header"
  >
    <div
      className="head"
    >
      <div
        className="name-container"
      >
        <h1
          className="name"
        >
          Process Name
        </h1>
      </div>
      <div
        className="tools"
      >
        <Link
          className="tool-button edit-button"
          to="edit"
        >
          <ForwardRef(Button)>
            <Icon
              type="edit"
            />
            Edit
          </ForwardRef(Button)>
        </Link>
        <ForwardRef(Button)
          className="tool-button delete-button"
          onClick={[Function]}
        >
          <Icon
            type="delete"
          />
          Delete
        </ForwardRef(Button)>
      </div>
    </div>
  </div>
  <Themed(BPMNDiagramErrorHandler)
    xml="process XML"
  >
    <ProcessRenderer
      mappings={Object {}}
    />
  </Themed(BPMNDiagramErrorHandler)>
  <DeleterErrorHandler
    deleteEntity={[Function]}
    entity={null}
    onClose={[Function]}
    onDelete={[MockFunction]}
    type="process"
  />
</div>
`;
